shared/types/indexer.go

```diff
@@ -0,0 +1,239 @@
+package types
+
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/jordanorelli/lexnum"
+	"github.com/pokt-network/pocket/persistence/kvstore"
+	"github.com/pokt-network/pocket/shared/crypto"
+)
+
+// TODO (Team) this is only going here in shared temporarily. It should go where the mepool goes (likely Utility module) in #163
+
+// Interfaces
+
+type TxIndexer interface { // TxIndexer interface defines methods to index and query transactions.
+	// Index analyzes, indexes and stores a single transaction result.
+	// Index, indexes by hash, height, sender, and recipient
+	Index(result TxResult) error
+
+	// GetByHash returns the transaction specified by hash or nil if the transaction is not indexed
+	GetByHash(hash []byte) (TxResult, error)
+
+	// GetByHeight returns the transactions specified by height or nil if there are no transactions at that height
+	GetByHeight(height int64) ([]TxResult, error)
+
+	// GetBySender returns the transactions signed by *sender* may be ordered descending/ascending
+	GetBySender(sender string, descending bool) ([]TxResult, error)
+
+	// GetByRecipient returns the transactions *sent to address* may be ordered descending/ascending
+	GetByRecipient(rec string, descending bool) ([]TxResult, error)
+
+	// Close stops the underlying db connection
+	Close() error
+}
+
+type TxResult interface {
+	GetTx() []byte                        // the transaction object primitive
+	GetHeight() int64                     // height it was sent
+	GetIndex() int32                      // which index it was within the block-transactions
+	GetResultCode() int32                 // 0 is no error, otherwise corresponds to error object code
+	GetError() string                     // can be empty
+	GetSender() string                    // get the address who signed
+	GetRecipient() string                 // can be empty
+	GetMessageType() string               // corresponds to type of message (Ex. validator-stake, app-unjail, node-stake) etc.
+	Hash() ([]byte, error)                // the hash of the tx bytes
+	HashFromBytes([]byte) ([]byte, error) // the hash of the tx bytes, avoids re-marshalling
+	Bytes() ([]byte, error)               // proto marshalled bytes
+	FromBytes([]byte) (TxResult, error)   // from proto marshalled bytes
+}
+
+var _ TxResult = &DefaultTxResult{}
```

Comment:
Are you referring to our deployment? Just because we have it there doesn't mean it doesn't have a place in the codebase during development. These strict checks saved me several times in the past from breaking interfaces

Reply:
`These strict checks saved me several times in the past from breaking interfaces`

+1 agreed about having them here (and elsewhere). 

I do not remember the context of my original comment so resolving.